### DISCUSSION TITLE
VSCode Compile Error - total coding noob

### DISCUSSION THREAD

**Discussion by mickstone** [picocomputer/community#104] [Q&A] [answered]
Hi All,

I'm trying to build the hello in the toolchain and get the following error. Any tips on where I need to look?
Looks like add_executable is set to "hello" but I'm not sure what to do with CMake configure.

Running on Ubuntu.

main] Building folder: PicoComputer hello
[build] Starting build
[proc] Executing command: /usr/bin/cmake --build /home/mick/Documents/PicoComputer/PicoComputer/build --config Debug --target hello -j 14 --
[build] [ 50%] Building C object CMakeFiles/hello.dir/src/main.c.obj
[build] cc: error: unrecognized command-line option ‘--add-source’; did you mean ‘--pad-source’?
[build] gmake[3]: *** [CMakeFiles/hello.dir/build.make:76: CMakeFiles/hello.dir/src/main.c.obj] Error 1
[build] gmake[2]: *** [CMakeFiles/Makefile2:100: CMakeFiles/hello.dir/all] Error 2
[build] gmake[1]: *** [CMakeFiles/Makefile2:107: CMakeFiles/hello.dir/rule] Error 2
[build] gmake: *** [Makefile:124: hello] Error 2
[proc] The command: /usr/bin/cmake --build /home/mick/Documents/PicoComputer/PicoComputer/build --config Debug --target hello -j 14 -- exited with code: 2
[driver] Build completed: 00:00:00.051
[build] Build finished with exit code 2
[main] =======================================================
[main] No executable target was found to launch. Please check:
[main]  - Have you called add_executable() in your CMake project?
[main]  - Have you executed a successful CMake configure?
[main] No program will be executed

**Comment by vruumllc** [comment]
I can't tell from the error messages above what procedure you are using to compile the project.

Are you trying to build from the command line, or are you using "Run Build Task..." (Ctrl+Shift+B) from the Terminal menu of vscode?

If you have everything set up per the instructions, and have selected RP6502 for your vscode build kit, then configuration happens automatically when you use vscode.
 
![vscode_cmake_kit_selection](https://github.com/picocomputer/community/assets/145504135/fcd731e9-ddbb-482a-982a-67119b934e4e)

**Reply by mickstone** [reply]
### I'm not sure. If I do Run Build Task I get the errors shown.
I'm sure I followed the instructions, but new to Ubuntu also, so this is a massive learning experience for me.

![Screenshot from 2024-04-27 11-03-22](https://github.com/picocomputer/community/assets/29742757/9e5fee59-604f-4b4a-ad9c-c8fb7cd513df)

**Reply by mickstone** [reply]
I'm thinking maybe I should start from scratch and build an Ubuntu VM. It would be easier to get back to a clean state if I bork something on the way.
What LTS stream would be best to use? 20, 22 or 24?

Thanks for helping :)

**Reply by vruumllc** [reply]
Everything looks good with your vscode and project settings. It is probably your compiler setup, then.

Before you go to the trouble of redoing Ubuntu, from an Ubuntu terminal command line, try typing these (my results shown):

tonyvr@tonyvr-laptop:~$ which cc65
/usr/bin/cc65

tonyvr@tonyvr-laptop:~$ cc65 --version
cc65 V2.19 - N/A

tonyvr@tonyvr-laptop:~$ env | grep PATH
PATH=/home/tonyvr/.cargo/bin:/home/tonyvr/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

tonyvr@tonyvr-laptop:~$ 

This is just to make sure the cc65 compiler is installed and can be found by vscode.

**Reply by mickstone** [reply]
Okay it looks like I hadn't done the path, but still getting the same error.

cc65 --version
cc65 V2.19 - Git b993d8833

path is
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin

I did a sudo make avail and pretty sure it can find the binaries via the symlinks in /usr/local/bin
I can run cc65 without providing a path no matter what directory I'm in.

This what I get when I run build task

[main] Building folder: PicoComputer hello
[build] Starting build
[proc] Executing command: /usr/bin/cmake --build /home/mick/Documents/PicoComputer/PicoComputer/build --config Debug --target hello -j 14 --
[build] [ 50%] Building C object CMakeFiles/hello.dir/src/main.c.obj
[build] cc: error: unrecognized command-line option ‘--add-source’; did you mean ‘--pad-source’?
[build] gmake[3]: *** [CMakeFiles/hello.dir/build.make:76: CMakeFiles/hello.dir/src/main.c.obj] Error 1
[build] gmake[2]: *** [CMakeFiles/Makefile2:100: CMakeFiles/hello.dir/all] Error 2
[build] gmake[1]: *** [CMakeFiles/Makefile2:107: CMakeFiles/hello.dir/rule] Error 2
[build] gmake: *** [Makefile:124: hello] Error 2
[proc] The command: /usr/bin/cmake --build /home/mick/Documents/PicoComputer/PicoComputer/build --config Debug --target hello -j 14 -- exited with code: 2
[driver] Build completed: 00:00:00.051
[build] Build finished with exit code 2
[main] =======================================================
[main] No executable target was found to launch. Please check:
[main]  - Have you called add_executable() in your CMake project?
[main]  - Have you executed a successful CMake configure?
[main] No program will be executed

**Reply by mickstone** [reply]
I built a VM during the day today and managed to get "hello" to compile. I think maybe my vscode got hung up somewhere and never configured the toolchain right.

Next on the list rebuilding my Ubuntu partition and working out how to get pyserial to install. I get the following error trying to install.

Running python3 -m pip install pyserial produces the error message below.

error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.
    
    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.
    
    See /usr/share/doc/python3.12/README.venv for more information.

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.

Omg, I wish some of this made sense ...sorry

**Reply by vruumllc** [reply]
Congrats! and what a relief. I had run out of suggestions. You seem to have done everything just fine, so I remain puzzled about the problem.

I have never encountered the pyserial problem, but this link seems to explain it all. Good luck!
https://www.baeldung.com/linux/pip-fix-externally-managed-environment-error

**Reply by vruumllc** [reply]
Reading the link above, I think you should be able to get pyserial installed using apt:

tonyvr@tonyvr-laptop:~$ sudo apt install python3-serial

**Reply by mickstone** [reply, marked as answer]
Thanks for your help :)

Looks like python3-serial is already installed so I think I make be good. A fresh install of Ubuntu 24.04 has fixed my issue.

Look out world, here I come. Or should I say "Hello, world!" here I come :)

Mick
